### DISCUSSION TITLE
[view] Component Props Type

### DISCUSSION THREAD

**Discussion by jin-Pro** [githru/githru-vscode-ext#56] [View]
저랑 성현님 께서 Detail Component를 담당하게 되었습니다.
View 팀 작업이 간단 컴포넌트 구현이라 간단하다고 생각하실수도 있겠지만,
그 중에서도 제일 간단한 Detail Component를 담당하게 되어서,,, ( D3 를 사용하지 않습니다,, )

때문에 Detail Component 구현 이외에도 

데이터 형식 맞추기
전체 환경 셋업
개발환경 문서
테스트

위 4가지 작업 중 몇가지를 진행해보면 좋겠다고 생각을 하였습니다. ( + 영택님의 의견을 반영하였습니다. )

---

제가 생각했을 때, 가장 필요한 작업이 각자 Component를 구현하다보니 

- 필요한 데이터가 다를것 ( 각자의 방식으로 데이터를 가공할 것으로 예상 )
- 사용하는 데이터에 따른 다양한 데이터 타입 ( 사용하는 데이터가 다르다보니 type을 여러개를 찍어 낼것으로 예상 )

위 두가지 이슈가 발생할 것 같습니다.

각자 컴포넌트를 구현하고나서, 7일에 병합하면서 맞춰보자고 얘기를 나눴던 것 같습니다.
좀 더 원활한 진행과 엔진팀에 요청드릴 내용을 미리 말해주어야할 것 같아서

Detail 팀에서 각 컴포넌트에서 필요한 데이터 형식과 타입을 정리해 보려 하는데 어떻게 생각하시나요?
그러기 위해서는 지금까지 구현해주신 Component에 대한 정보가 필요할 것 같습니다!

**Comment by jin-Pro** [comment]
31일 오프 모임에서

데이터 형식 맞추기
전체 환경 셋업
개발환경 문서
테스트

에 대한 이야기도 진행해보고 싶습니다!

**Reply by hanseul-lee** [reply]
좋습니다~ 너무 당연한 부분이고 시작할때부터 논의되었으면 좋았을 내용이라고 생각합니다

**Comment by hanseul-lee** [comment]
- Statistics - Bar

```ts
export type AuthorDataType = {
  name: string;
  commit: number;
  deletion: number;
  insertion: number;
};
```

Statistics 내 bar chart에서 사용하는 데이터 형식 및 타입입니다.
샘플 데이터의 `ClusterNode[]`에서 가공해 `AuthorDataType[]` 으로 받아와 사용하고 있습니다.
이미 가공해서 사용하고 있기에 엔진팀 추가로 작업해주셔도 좋고 지금 사용하는 그대로 써도 큰 영향 없습니다

**Comment by jejecrunch** [comment]
* Vertical Clust List

저희는 이슈에 답변으로 달았었기 때문에 링크로 대체하겠습니다.
네이밍의 경우 내일 변경될 수도 있을 것 같습니다.
저희의 경우도 statistics 팀과 마찬가지로 sample data를 가공해서 사용하고 있습니다.

https://github.com/githru/githru-vscode-ext/issues/40#issuecomment-1231700616
